packages/compass-data-modeling/src/components/drawer/collection-drawer-content.tsx

```diff
@@ -70,17 +74,77 @@ const relationshipContentStyles = css({
   marginTop: spacing[400],
 });
 
+export function getIsCollectionNameValid(
+  collectionName: string,
+  namespaces: string[],
+  namespace: string
+): {
+  isValid: boolean;
+  errorMessage?: string;
+} {
+  if (collectionName.trim().length === 0) {
+    return {
+      isValid: false,
+      errorMessage: 'Collection name cannot be empty.',
+    };
+  }
+
+  const namespacesWithoutCurrent = namespaces.filter((ns) => ns !== namespace);
+
+  const isDuplicate = namespacesWithoutCurrent.some(
+    (ns) =>
+      ns === `${toNS(namespace).database}.${collectionName}` ||
+      ns === `${toNS(namespace).database}.${collectionName.trim()}`
+  );
+
+  return {
+    isValid: !isDuplicate,
+    errorMessage: isDuplicate ? 'Collection name must be unique.' : undefined,
+  };
+}
+
 const CollectionDrawerContent: React.FunctionComponent<
   CollectionDrawerContentProps
 > = ({
   namespace,
+  namespaces,
+  note = '',
   relationships,
   onCreateNewRelationshipClick,
   onEditRelationshipClick,
   onDeleteRelationshipClick,
-  note = '',
   onNoteChange,
+  onRenameCollection,
 }) => {
+  const [collectionName, setCollectionName] = useState(
+    () => toNS(namespace).collection
+  );
+
+  const {
+    isValid: isCollectionNameValid,
+    errorMessage: collectionNameEditErrorMessage,
+  } = useMemo(
+    () => getIsCollectionNameValid(collectionName, namespaces, namespace),
+    [collectionName, namespaces, namespace]
+  );
+
+  useLayoutEffect(() => {
+    setCollectionName(toNS(namespace).collection);
+  }, [namespace]);
+
+  const onBlurCollectionName = useCallback(() => {
```

Comment:
Just FYI, added a shared hook for that in another PR when adding notes https://github.com/mongodb-js/compass/pull/7171#discussion_r2259611500